skrf/network.py

```diff
@@ -1330,104 +1330,121 @@ def n(self) -> npy.ndarray:
-      """
-      the frequency vector for the noise of the network, in Hz.
-      """
-      if not self.noisy:
-        raise ValueError('network does not have noise')
-      return self.noise_freq
+        """
+        the frequency vector for the noise of the network, in Hz.
+        """
+        if not self.noisy:
+            raise ValueError('network does not have noise')
+        return self.noise_freq
 
     @property
     def y_opt(self) -> npy.ndarray:
-      """
-      the optimum source admittance to minimize noise
-      """
-      noise = self.n
-      return (npy.sqrt(noise[:,1,1]/noise[:,0,0] - npy.square(npy.imag(noise[:,0,1]/noise[:,0,0])))
-          + 1.j*npy.imag(noise[:,0,1]/noise[:,0,0]))
+        """
+        the optimum source admittance to minimize noise
+        """
+        noise = self.n
+        return (npy.sqrt(noise[:, 1, 1] / noise[:, 0, 0] - npy.square(npy.imag(noise[:, 0, 1] / noise[:, 0, 0])))
+                + 1.j * npy.imag(noise[:, 0, 1] / noise[:, 0, 0]))
 
     @property
     def z_opt(self) -> npy.ndarray:
-      """
-      the optimum source impedance to minimize noise
-      """
-      return 1./self.y_opt
+        """
+        the optimum source impedance to minimize noise
+        """
+        return 1. / self.y_opt
 
     @property
     def g_opt(self) -> npy.ndarray:
-      """
-      the optimum source reflection coefficient to minimize noise
-      """
-      return z2s(self.z_opt.reshape((self.f.shape[0], 1, 1)), self.z0[:,0])[:,0,0]
+        """
+        the optimum source reflection coefficient to minimize noise
+        """
+        return z2s(self.z_opt.reshape((self.f.shape[0], 1, 1)), self.z0[:, 0])[:, 0, 0]
 
     @property
     def nfmin(self) -> npy.ndarray:
-      """
-      the minimum noise figure for the network
-      """
-      noise = self.n
-      return npy.real(1. + (noise[:,0,1] + noise[:,0,0] * npy.conj(self.y_opt))/(2*K_BOLTZMANN*T0))
+        """
+        the minimum noise figure for the network
+        """
+        noise = self.n
+        return npy.real(1. + (noise[:, 0, 1] + noise[:, 0, 0] * npy.conj(self.y_opt)) / (2 * K_BOLTZMANN * T0))
 
     @property
     def nfmin_db(self) -> npy.ndarray:
-      """
-      the minimum noise figure for the network in dB
-      """
-      return mf.complex_2_db10(self.nfmin)
-
-    def nf(self, z: NumberLike) -> npy.ndarray:
-      """
-      the noise figure for the network if the source impedance is z
-      """
-      z0 = self.z0
-      y_opt = self.y_opt
-      fmin = self.nfmin
-      rn = self.rn
-
-      ys = 1./z
-      gs = npy.real(ys)
-      return fmin + rn/gs * npy.square(npy.absolute(ys - y_opt))
+        """
+        the minimum noise figure for the network in dB
+        """
+        return mf.complex_2_db10(self.nfmin)
+
+    def nf(self, z: NumberLike, passive=False, Ta=290, Tp=290) -> npy.ndarray:
+        """
+        the noise figure for the network if the source impedance is z
```

Comment:
Could you add a few words and a reference to the theorem you use to calculate the NF if not defined in the ntwk?